Editor/EditorCore/UVEditor.cs

```diff
@@ -718,44 +718,81 @@ void OnScreenResize()
             Repaint();
         }
 
+        bool IsCopyUVSettingsModifiers(EventModifiers modifiers)
+        {
+            return (modifiers == (EventModifiers.Control | EventModifiers.Shift));
+        }
+
         /**
-         * return true if shortcut should eat the event
+         * returns true if a copy of UV settingsoccured from the first selected face to the target face
          */
-        internal bool ClickShortcutCheck(ProBuilderMesh pb, Face selectedFace)
+        bool CopyFaceUVSettings(ProBuilderMesh pb, Face targetFace)
         {
-            Event e = Event.current;
+            // get first selected UV face
+            ProBuilderMesh firstObj;
+            Face[] source = new Face[1];
 
-            // Copy UV settings
-            if (e.modifiers == (EventModifiers.Control | EventModifiers.Shift))
+            ProBuilderEditor.instance.GetFirstSelectedFace(out firstObj, out source[0]);
```

Comment:
I think this should instead check `MeshSelection.activeMesh` and `MeshSelection.activeFace`.

Reply:
Changed